jekyll/_posts/2021-09-25-version-160-released.md

```diff
@@ -0,0 +1,817 @@
+* made JS, VM and nimscript backend more consistent with C backend, allowing more modules to work with those backends, including the imports from `std/prelude`; the test suite now standardizes on testing stdlib modules on each major backend (C, JS, VM).
+* support for Apple silicon/M1, 32-bit RISC-V, CROSSOS, improved support for NodeJS backend
+* major improvements to the following modules: `system, math, random, json, jsonutils, os, typetraits, wrapnils, lists, hashes` including performance improvements
+* deprecated a number of error prone or redundant mis-features
+
+
+# Installing Nim 1.6
+We recommend everyone to upgrade to 1.6:
+
+## New users
+
+Check out if the package manager of your OS already ships version 1.6 or
+install it as described [here](https://nim-lang.org/install.html).
+
+
+## Existing users
+
+If you have installed a previous version of Nim using `choosenim`,
+getting Nim 1.6 is as easy as:
+
+```bash
+choosenim update self
+choosenim update stable
+```
+
+If you don't have `choosenim`, you can follow the same
+[install link](https://nim-lang.org/install.html) as above.
+
+## Building from source
+```bash
+git clone https://github.com/nim-lang/Nim
+cd Nim
+sh build_all.sh
+```
+The last command can be re-run after pulling new commits.
+Note that the `csources` repo used was changed to `csources_v1`, the new setup is designed to be forward and backward compatible.
+
+## Building from a CI setup
+We now have an API to be used in CI which abstracts the implementation details: `. ci/funs.sh && nimBuildCsourcesIfNeeded`; in fact all the existing CI pipelines have been refactored to use this, see
+[#17815](https://github.com/nim-lang/Nim/pull/17815).
+
+
+# Contributors to Nim 1.6
+
+Many thanks to our recurring and new [contributors](https://github.com/nim-lang/Nim/graphs/contributors?from=2020-10-16&to=2021-09-25&type=c),
+Nim is a community driven collaborative effort that welcomes all contributions, big or small.
+
+
+# Backward compatibility and preview flags
+
+Starting with this release, we've introduced preview flags of the form `-d:nimPreviewX`
+(e.g. `-d:nimPreviewFloatRoundtrip`); they allow users to opt-in a new stdlib/compiler behavior
+that will likely become default in the next or a future release.
+These staging flags are aimed at minimizing backward compatibility issues.
+We also introduced opt-out flags of the form `-d:nimLegacyX`,  e.g. `-d:nimLegacyCopyFile`,
+for cases where the default was changed to the new behavior,
+and an explicit opt-out is needed for some transition period.
+
+# Major new features
+With so many new features, pinpointing the most salient ones is a subjective exercise,
+but here are a select few:
+
+
+## Strict effects
+The effect system was refined and there is a new `.effectsOf` annotation that does
+  explicitly what was previously done implicitly. See the manual for details.
```

Comment:
Link here for the manual would be awesome :)

Reply:
done